src/App.tsx

```diff
@@ -1,27 +1,109 @@
-import React from 'react';
+import React, { useState } from 'react';
+import classNames from 'classnames';
+import 'bulma/css/bulma.min.css';
 import './App.scss';
 import { GoodsList } from './GoodsList';
+import { Good } from './types/Good';
 
-// import { getAll, get5First, getRed } from './api/goods';
-// or
-// import * as goodsAPI from './api/goods';
+import {
+  getAll,
+  get5First,
+  getRedGoods,
+  wait,
+} from './api/goods';
+import { LoadingError } from './components/LoadingError';
 
-export const App: React.FC = () => (
-  <div className="App">
-    <h1>Dynamic list of Goods</h1>
+export const App: React.FC = () => {
+  const [goods, setGoods] = useState<Good[]>([]);
+  const [loading, setLoading] = useState(false);
+  const [loadingFirstFive, setLoadingFirstFive] = useState(false);
+  const [loadingRed, setLoadingRed] = useState(false);
```

Comment:
Let's disable buttons when something loading instead, it's strange that we have 3 states for loading and that we can make request while we didn't get previous result